src/smc-webapp/assistant/main.cjsx

```diff
@@ -0,0 +1,438 @@
+        cat1                : rtypes.number      # index of selected second category (second from left)
+        cat2                : rtypes.number      # index of selected third category (document titles)
+        catlist0            : rtypes.arrayOf(rtypes.string)  # list of first category entries
+        catlist1            : rtypes.arrayOf(rtypes.string)  # list of second level categories
+        catlist2            : rtypes.arrayOf(rtypes.string)  # third level are the document titles
+        code                : rtypes.string      # displayed content of selected document
+        setup_code          : rtypes.string      # optional, common code in the sub-category
+        prepend_setup_code  : rtypes.bool        # if true, setup code is prepended to code
+        descr               : rtypes.string      # markdown-formatted content of document description
+        hits                : rtypes.arrayOf(rtypes.array)  # search results
+        search_str          : rtypes.string      # substring to search for -- or undefined
+        search_sel          : rtypes.number      # index of selected matched documents
+        submittable         : rtypes.bool        # if true, the buttons at the bottom are active
+        cat1_top            : rtypes.arrayOf(rtypes.string)
+        unknown_lang        : rtypes.bool        # true if there is no known set of documents for the language
+
+    getInitialState: ->
+        INIT_STATE
+
+    data_lang: ->
+        @get('data').get(@get('lang'))
+
+    # First categories list, depends on selected language, sort order depends on cat1_top
+    get_catlist0: () ->
+        cat0 = @data_lang().keySeq().toArray()
+        top = @get('cat1_top')
+        cat0ordering = (el) ->
+            i = - top.reverse().indexOf(el)
+            return [i, el]
+        return _.sortBy(cat0, cat0ordering)
+
+    # Second level categories list, depends on selected index of first level
+    # Sorted by cat1_top and a possible 'sortweight'
+    get_catlist1: () ->
+        k0 = @get_catlist0()[@get('cat0')]
+        cat1data = @data_lang().get(k0)
+        cat1 = cat1data.keySeq().toArray()
+        top = @get('cat1_top')
+        cat1ordering = (el) ->
+            so = cat1data?.getIn([el, 'sortweight']) ? 0.0
+            i = - top.reverse().indexOf(el)
+            return [so, i, el]
+        return _.sortBy(cat1, cat1ordering)
+
+    # The titles of the selected documents are exactly as they're in the original data (they're an array)
+    # That way, it's possible to create a coherent narrative from top to bottom
+    get_catlist2: () ->
+        k0 = @get_catlist0()[@get('cat0')]
+        k1 = @get_catlist1()[@get('cat1')]
+        return @data_lang().getIn([k0, k1, 'entries']).map((el) -> el.get(0)).toArray()
+
+
+class ExamplesActions extends Actions
+    _init: (store) ->
+        @store = store
+
+    get: (key) ->
+        @store.get(key)
```

Comment:
For top level store items using the newer store construction method you can just do `@store.prepend_setup_code` for items defined in stateTypes. But I think @williamstein doesn't like that as much?